components/tenant-fetcher/cmd/main.go

```diff
@@ -65,29 +87,65 @@ func main() {
 		exitOnError(errors.New("missing handler endpoint or tenant path parameter"), "Error while loading app handler config")
 	}
 
+	transact, closeFunc, err := persistence.Configure(ctx, cfg.Database)
+	exitOnError(err, "Error while establishing the connection to the database")
+
+	authenticatorsConfig, err := authenticator.InitFromEnv(envPrefix)
+	exitOnError(err, "Failed to retrieve authenticators config")
+
+	defer func() {
+		err := closeFunc()
+		exitOnError(err, "Error while closing the connection to the database")
+	}()
+
+	uidSvc := uuid.NewService()
+	converter := tenant.NewConverter()
+	repo := tenant.NewRepository(converter)
+	service := tenant.NewService(repo, transact, uidSvc)
+
 	ctx, err = log.Configure(ctx, &cfg.Log)
 	exitOnError(err, "Failed to configure Logger")
 	logger := log.C(ctx)
 
+	middleware := auth.New(
+		cfg.JwksEndpoint,
+		cfg.IdentityZone,
+		cfg.SubscriptionCallbackScope,
+		extractTrustedIssuersScopePrefixes(authenticatorsConfig),
+		cfg.AllowJWTSigningNone,
+	)
+
+	logger.Infof("JWKS synchronization enabled. Sync period: %v", cfg.JWKSSyncPeriod)
+	periodicExecutor := executor.NewPeriodic(cfg.JWKSSyncPeriod, func(ctx context.Context) {
+		err := middleware.SynchronizeJWKS(ctx)
+		if err != nil {
+			logger.WithError(err).Error("An error has occurred while synchronizing JWKS")
+		}
+	})
+	go periodicExecutor.Run(ctx)
+
 	mainRouter := mux.NewRouter()
 	subrouter := mainRouter.PathPrefix(cfg.RootAPI).Subrouter()
+	subrouter.Use(middleware.Handler())
 
 	logger.Infof("Registering Tenant Onboarding endpoint on %s...", cfg.HandlerEndpoint)
-	subrouter.HandleFunc(cfg.HandlerEndpoint, getOnboardingHandlerFunc(cfg.TenantPathParam)).Methods(http.MethodPut)
+	subrouter.HandleFunc(cfg.HandlerEndpoint, getOnboardingHandlerFunc(service, cfg.TenantPathParam, cfg.TenantProviderAccountId)).Methods(http.MethodPut)
```

Comment:
The two onboarding and decommissioning handler functions can be part of a Handler struct which can be part of the `tenant` package (like [here](https://github.com/kyma-incubator/compass/blob/master/components/connectivity-adapter/internal/appregistry/service/handler.go))